exe/Benchmarks.hs

```diff
@@ -0,0 +1,33 @@
+module Benchmarks (main) where
+
+import           Protolude
+
+import           Criterion.Main
+
+import           Radicle
+
+steps :: [Text] -> Value
+steps is = runIdentity $ snd <$> foldlM step (pureEnv, Number 0) is
+  where
+    step (s,_) t = do
+      (res_, s') <- interpretWithState "[bench]" t s
+      case res_ of
+        Left e  -> panic (show e)
+        Right v -> pure (s', v)
+
+counter :: Int -> Rational
+counter i = unwrap $ steps $
+  [ "(def i (ref 0))"
+  , "(def add (fn [x] (write-ref i (+ (read-ref i) x))))"
+  , "(def eval (fn [expr state] (base-eval (list 'add expr) state)))" ]
+  ++ (show <$> take i (cycle [1..20::Int]))
```

Comment:
Any particular reason to not do just `take i [1..]`?

Reply:
I just wanted the number in the ref to grow linearly rather than quadratically, didn't want to benchmark the performance of rational number addition on big numbers.

Probably completely unnecessary :smiley: